backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/AggregationDAO.java

```diff
@@ -129,7 +131,33 @@ public int findC_Invoice_Candidate_HeaderAggregationKey_ID(I_C_Invoice_Candidate
 		headerAggregationKeyRecord.setC_BPartner_ID(bpartnerId);
 		headerAggregationKeyRecord.setIsSOTrx(ic.isSOTrx());
 		headerAggregationKeyRecord.setIsActive(true);
-		InterfaceWrapperHelper.save(headerAggregationKeyRecord);
+		saveHeaderAggregationKeyWithRetryOnFail(headerAggregationKeyRecord);
 		return headerAggregationKeyRecord.getC_Invoice_Candidate_HeaderAggregation_ID();
 	}
+
+	private void saveHeaderAggregationKeyWithRetryOnFail(@NonNull final I_C_Invoice_Candidate_HeaderAggregation headerAggregationKeyRecord)
```

Comment:
what I had in mind was something like:

- try catch around the save action
  - if it fails with `DBUniqueConstraintException` then try to do a new lookup, i.e. 
  ```JAVA
   // Find existing header aggregation key ID
		final int headerAggregationKeyId = Services.get(IQueryBL.class)
				.createQueryBuilder(I_C_Invoice_Candidate_HeaderAggregation.class, ic)
				.addEqualsFilter(I_C_Invoice_Candidate_HeaderAggregation.COLUMN_HeaderAggregationKey, headerAggregationKeyCalc)
				.create()
				.firstIdOnly();
   ```
     - side-note: pls also put this into a method
   - if we still can't find a record for our aggregation key, then throw the exception further